test_common/harness/imageHelpers.cpp

```diff
@@ -796,7 +796,8 @@ void get_max_sizes(
 
     (*numberOfSizes) = 0;
 
-    if (image_type == CL_MEM_OBJECT_IMAGE1D)
+    if (image_type == CL_MEM_OBJECT_IMAGE1D
+        || image_type == CL_MEM_OBJECT_IMAGE1D_BUFFER)
```

Comment:
It looks like ln:806 isn't taking into account pixel size and is assigning the entire maxWidth value as the maxSize for the image 1D buffer.